src/fetchers/git-fetcher.js

```diff
@@ -73,7 +73,17 @@ export default class GitFetcher extends BaseFetcher {
     await git.initRemote();
     await git.clone(this.dest);
 
-    let tarballInMirrorPath = this.config.getOfflineMirrorPath(this.reference);
+    // Get the tarball filename from the url
+    const {pathname} = url.parse(this.reference);
+    let tarballFilename;
+    if (pathname == null) {
+      tarballFilename = this.reference;
+    } else {
+      tarballFilename = path.basename(pathname);
+    }
+
+    let tarballInMirrorPath = this.config.getOfflineMirrorPath(tarballFilename);
```

Comment:
Sorry, I missed that this only breaks the scoped packages users and yeah, I agree it should be fine to fix this.